tests/unit/FileHelperTest.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+namespace WP2Static;
+
+use Mockery;
+use org\bovigo\vfs\vfsStream;
+use WP_Mock;
+use WP_Mock\Tools\TestCase;
+
+final class FileHelperTest extends TestCase {
+    public function setUp() : void {
+        WP_Mock::setUp();
+    }
+
+    public function tearDown() : void {
+        WP_Mock::tearDown();
+    }
+
+    /**
+     * Test delete_dir_with_files method
+     *
+     * @return void
+     */
+    public function testDeleteDirWithFiles() {
+        // Set up a virual folder structure
+        $structure = [
+            'folder_1' => [
+                'file_1.txt' => 'file_1.txt',
+                'file_2.txt' => 'file_2.txt',
+            ],
+            'folder_2' => [
+                'file_3.txt' => 'file_3.txt',
+                'file_4.txt' => 'file_4.txt',
+                'folder_3' => [
+                    'file_5.txt' => 'file_5.txt',
+                    'file_6.txt' => 'file_6.txt',
+                    
+                ],
+            ],
+        ];
+        $vfs = vfsStream::setup('root');
+        vfsStream::create($structure, $vfs);
+        $filepath = vfsStream::url('root/folder_1');
+
+        // Check vfsStream set up the directories correctly
+        $expected = true;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // folder_1 should now be gone
+        FilesHelper::delete_dir_with_files($filepath);
+        $expected = false;
+        $actual = is_dir( $filepath );
+        $this->assertEquals( $expected, $actual );
+
+        // Delete a nested folder
+        $filepath = vfsStream::url('root/folder_2/folder_3');
```

Comment:
May as well add an extra assertion here to ensure the nested dir was deleted?

Reply:
If the parent folder was deleted there's no possible way the child folder could exist. The assertion wouldn't really be useful here I think.

Reply:
Sometimes (often), simple logic alludes me, sorry!

Reply:
(evades?), both logic and words, it seems!

Reply:
I'm adding responses as I work on your queries/concerns. If you're happy with my explanations or changes make sure to hit resolve so I know nothing more needs to be done. You responded positively to this one but it seems rude to hit Resolve myself when it wasn't me who brought up the issue.

Reply:
Ah, thanks for alerting me! Resolving!